File: backend/app/schemas/stakeholder.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List
from decimal import Decimal
from ..models import StakeholderType, TrustZone


class StakeholderBase(BaseModel):
    name: str
    type: StakeholderType
    email: Optional[str] = None
    phone: Optional[str] = None
    address: Optional[str] = None
    district_id: Optional[int] = None
    block_id: Optional[int] = None


class StakeholderCreate(StakeholderBase):
    pass


class StakeholderUpdate(BaseModel):
    name: Optional[str] = None
    type: Optional[StakeholderType] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    address: Optional[str] = None
    district_id: Optional[int] = None
    block_id: Optional[int] = None
    is_active: Optional[bool] = None


class TrustScoreResponse(BaseModel):
    id: int
    stakeholder_id: int
    score: Decimal
    zone: TrustZone
    delivery_performance: Decimal
    quality_compliance: Decimal
    grievance_rate: Decimal
    data_accuracy: Decimal
    calculated_at: datetime

    class Config:
        from_attributes = True


class StakeholderResponse(StakeholderBase):
    id: int
    is_active: bool
    created_at: datetime
    updated_at: Optional[datetime]
    trust_scores: Optional[List[TrustScoreResponse]] = None

    class Config:
        from_attributes = True


class StakeholderListResponse(BaseModel):
    items: List[StakeholderResponse]
    total: int
    page: int
    page_size: int

File: frontend/src/types/stakeholder.ts
export type StakeholderType = 
  | 'SUPPLIER'
  | 'TRANSPORT_FLEET'
  | 'AWW'
  | 'CDPO'
  | 'SUPERVISOR'
  | 'COLLECTOR'
  | 'SECRETARY'

export type TrustZone = 'GREEN' | 'YELLOW' | 'ORANGE' | 'RED'

export interface Stakeholder {
  id: number
  name: string
  type: StakeholderType
  email?: string
  phone?: string
  address?: string
  district_id?: number
  block_id?: number
  is_active: boolean
  created_at: string
  updated_at?: string
  trust_scores?: TrustScore[]
}

export interface TrustScore {
  id: number
  stakeholder_id: number
  score: number
  zone: TrustZone
  delivery_performance: number
  quality_compliance: number
  grievance_rate: number
  data_accuracy: number
  calculated_at: string
}

export interface StakeholderListResponse {
  items: Stakeholder[]
  total: number
  page: number
  page_size: number
}
